doc/lsst.pipe.base/command-line-task-data-repo-howto.rst

```diff
@@ -0,0 +1,136 @@
+--------------------------------
+
+The output repository for one task can become the input repository for the next command-line task.
+For example:
+
+.. code-block:: text
+
+   task2.py outputrepo --output outputrepo2 ...
+
+Because Butler data repositories are *chained*, the output repository (here, :file:`outputrepo2`) will appear to contain all the datasets from the input repositories (here: :file:`inputrepo`, :file:`outputrepo`, and :file:`outputrepo2` itself).
+
+How to re-use output repositories
+---------------------------------
+
+An output repository can be the same as the input repository:
+
+.. code-block:: text
+
+   task3.py outputrepo2 --output outputrepo2 ...
+
+This pattern is useful for reducing the number of repository directories on a filesystem.
+Packing outputs from multiple tasks into one output repository does reduce your flexibility to run a task several times with different configurations and compare outputs, though.
+
+You can also run the same task multiple times with the same output repository.
+Be aware that the Science Pipelines will help you maintain the integrity of the processed data's provenance.
+If you change a task's configuration and re-run the task into the same output repository, an error "Config does not match existing task config" will be shown.
+
+How to use repository path environment variables
+------------------------------------------------
+
+The :envvar:`PIPE_INPUT_ROOT` and :envvar:`PIPE_OUTPUT_ROOT` environment variables help you specify data repository paths more succinctly.
+When set, the :option:`REPOPATH` argument path is treated as relative to :envvar:`PIPE_INPUT_ROOT` and the :option:`--output` path is relative to :envvar:`PIPE_OUTPUT_ROOT`.
+See :ref:`command-line-task-envvar` for details.
+
+.. _command-line-task-data-repo-using-reruns:
+
+Using reruns to organize outputs in a single data repository
+============================================================
+
+An alternative way to organize output data repositories is with **reruns**.
+Reruns are a convention for repositories that are located within a single root data repository, rather than being spread across independent data repository directories.
+Reruns named directories found in the root data repository's (:option:`REPOPATH`) :file:`rerun` directory:
+
+.. code-block:: text
+
+   REPOPATH/rerun/rerun_name
+
+Being self-contained within a single root data repository directory, reruns tend to be easier to manage.
+
+.. seealso::
+
+  :option:`The --rerun argument's syntax reference <--rerun>`.
+
+How to create a rerun
+---------------------
+
+To use input data from a :file:`DATA` Butler repository and write outputs to a rerun called ``A``, set a command-line task's :option:`REPOPATH` and :option:`--rerun` like this:
+
+.. code-block:: bash
+
+   task1.py DATA --rerun A ...
+
+The rerun will be located inside the :file:`DATA` repository at :file:`DATA/rerun/A`.
+
+How to use one rerun as input to another (chaining)
```

Comment:
Please explicitly call this "rerun chaining" and not just "chaining," because the latter is reminiscent of how all repos have parents and are also "chained" in a sense.